python/cugraph/cugraph/utilities/utils.py

```diff
@@ -549,3 +553,13 @@ def create_directory_with_overwrite(directory):
     if os.path.exists(directory):
         shutil.rmtree(directory)
     os.makedirs(directory)
+
+
+def install_ssl_cert():
```

Comment:
Putting this here in `cugraph.utilities` means that now `certifi` needs to be introduced as a runtime dependency of `cugraph`.

> ModuleNotFoundError: No module named 'certifi'

([build link](https://github.com/rapidsai/cugraph/actions/runs/12265067332/job/34222694003?pr=4825))

I'm really nervous about the idea of introducing a new runtime dependency this long after code freeze and this close to the release.

And even just in general... does this really need to be in the `cugraph` package? Are those code paths in `datasets.py` intended to be used by downstream users, or are they just in the `cugraph` package for convenience in its own testing (and testing for `cugraph-gnn` libraries)?

Here in 24.12, to minimize risk, I think we should only run this at test time in CI here.

That'd probably mean:

* make `certifi` a test-only dependency
* put this bit of Python code in a script like `ci/install-certifi-certs.py`
* run that in each `ci/test_*` script, before any tests

Reply:
I agree that it is not a good time to add an additional runtime dependency..

The codepaths in `datasets.py` is used both by end users and for our own testing purposes. You can think of them as analogous to [NetworkX's built-in graphs](https://networkx.org/documentation/stable/auto_examples/graph/index.html)

I don't think that the SSL issue would affect our users. I think this is just affecting CI because of how our images are setup. It would probably be fine to go with the solution you suggested and install it for our tests. I'll go ahead and add that

Reply:
Ok got it, thanks for the explanation! Then yeah, I think doing this only as a testing dependency is a good solution.

I tested whether this could just be run outside of of `pytest`, like I suggested... and you're absolutely right, it can't. I guess something in this solution must also modify `os.environ` or similar.

<details><summary>code I used to confirm that (click me)</summary>

```shell
docker run \
    --rm \
    -it rapidsai/citestwheel:cuda11.8.0-rockylinux8-py3.10 \
    bash

python -c 'import urllib.request; urllib.request.urlretrieve("https://data.rapids.ai/cugraph/results/resultsets.tar.gz", "foo.tar.gz")'
# urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED

cat > ./install-ca-certs.py <<EOF
import certifi
from ssl import create_default_context
from urllib.request import build_opener, HTTPSHandler, install_opener

ssl_context = create_default_context(cafile=certifi.where())
https_handler = HTTPSHandler(context=ssl_context)
install_opener(build_opener(https_handler))
EOF

python -m pip install certifi
python ./install-ca-certs.py

python -c 'import urllib.request; urllib.request.urlretrieve("https://data.rapids.ai/cugraph/results/resultsets.tar.gz", "foo.tar.gz")'
# urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED
```

</details>

Given that, I think the way you've set this up as a testing-only dependency is a good solution. Thanks for working through it with me.

Reply:
Thank _you_ for helping track this down and brainstorming a solution with me.